### ISSUE TITLE
require('psi'): Error connecting to http://localhost:$PORT/insights (ECONNREFUSED)

### ISSUE THREAD

**Issue by emirotin** [bevacqua/perfschool#4]
As soon as I add `psi` as a required dependency I start getting `✗ Error connecting to http://localhost:50780/insights (ECONNREFUSED)` (the port number randomly changes, of course).

Here's the reference `solution.js`:

```
'use strict';

var fs = require('fs');
var path = require('path');
var express = require('express');
var psi = require('psi');

var app = express();
var port = process.env.PORT || 7777;

app.get('/', home);
app.get('/insights', insights);
app.listen(port, listening);

function listening () {
  console.log('Listening on port', port);
}

function home (req, res) {
  var file = path.join(__dirname, 'index.html');
  var index = fs.readFileSync(file, 'utf8');
  res.send(index);
}

function insights (req, res) {
  res.end('Not very insightful!');
}
```

As soon as I comment the PSI line I get

```
perfschool run solution.js                                                     [0:14:12]

# /insights produces valuable information about page weight
✓ Returns a 200 OK status code
✗ Returns a JSON object
✗ Response has a "resources" property
✗ Response has a "resources.js" property
✗ Response has a "resources.css" property
✗ Response has a "resources.total" property
✗ Response has a "resources.hosts" property
```

As soon as uncomment — the reported error.

**Comment by emirotin** [comment]
Also the PSI response format has changed, here's the recent one:

```
{
numberResources: 18,
numberHosts: 6,
totalRequestBytes: "1986",
numberStaticResources: 14,
htmlResponseBytes: "2959",
cssResponseBytes: "175745",
imageResponseBytes: "1342803",
javascriptResponseBytes: "230129",
otherResponseBytes: "755",
numberJsResources: 3,
numberCssResources: 3
}
```

(I got it by manually running the proper solution.js with node and copying the result from the browser.)

**Comment by bevacqua** [comment]
Could you create a PR fixing this issue?

**Comment by emirotin** [comment]
Unfortunately I'm not sure how to fix it, the format change doesn't look like the main source of the problem, see the 1st message

**Comment by psdcoder** [comment]
Hi! I also faced with this problem (ECONNREFUSED). 
After examining the code a bit, I came to the conclusion that solution server doesn't have time to run.
Timeout for calling test method I'm found in [workshopper-wat](https://github.com/bevacqua/workshopper-wat/blob/master/wat.js#L43) module. I just increase this value and the problem was solved.
Maybe it's because I'm set up tunnel globally, not in test method, and it takes some time.
